statistics/handle/statscache.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handle
+
+import (
+	"encoding/binary"
+	"sync"
+
+	"github.com/pingcap/tidb/statistics"
+	"github.com/pingcap/tidb/util/kvcache"
+	"github.com/pingcap/tidb/util/memory"
+)
+
+// statsCache caches table statistics.
+type statsCache struct {
+	mu          sync.Mutex
```

Comment:
it's better to use `RWMutex` to mitigate lock contention for reading operations.

Reply:
but in simplelcache(lru), read option also change the list data of cache.